packages/devtools_app/lib/src/flutter/banner_messages.dart

```diff
@@ -0,0 +1,304 @@
+
+  final _refreshMessagesController = StreamController<Null>.broadcast();
+
+  Stream<Null> get onRefreshMessages => _refreshMessagesController.stream;
+
+  @override
+  void dispose() {
+    _refreshMessagesController.close();
+  }
+
+  void refreshMessages() {
+    _refreshMessagesController.add(null);
+  }
+
+  void dismissMessage(String messageId) {
+    assert(!isMessageDismissed(messageId));
+    _dismissedMessageIds.add(messageId);
+    refreshMessages();
+  }
+
+  bool isMessageDismissed(String messageId) {
+    return _dismissedMessageIds.contains(messageId);
+  }
+}
+
+class BannerMessageContainer extends StatefulWidget {
+  const BannerMessageContainer({Key key, @required this.screen})
+      : super(key: key);
+
+  final Screen screen;
+
+  @override
+  _BannerMessageContainerState createState() => _BannerMessageContainerState();
+}
+
+class _BannerMessageContainerState extends State<BannerMessageContainer>
+    with AutoDisposeMixin {
+  BannerMessagesController controller;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final newController = Controllers.of(context)?.bannerMessages;
+    if (newController == controller) return;
+    controller = newController;
+
+    autoDispose(controller.onRefreshMessages.listen((_) => setState(() {})));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final messagesForScreen = widget.screen.messages(context);
+    if (messagesForScreen.isNotEmpty) {
+      final messagesToShow = <Widget>[];
+      for (Widget message in messagesForScreen) {
+        assert(message is UniqueMessage);
+        if (!controller.isMessageDismissed((message as UniqueMessage).id)) {
+          messagesToShow.add(message);
+        }
+      }
+      if (messagesToShow.isNotEmpty) {
+        return Column(
+          children: messagesToShow,
+        );
+      }
+    }
+    return const SizedBox();
+  }
+}
+
+class BannerMessage extends StatelessWidget implements UniqueMessage {
+  const BannerMessage({
+    @required this.messageId,
+    @required this.textSpans,
+    @required this.backgroundColor,
+    @required this.foregroundColor,
+  });
+
+  final String messageId;
+  final List<TextSpan> textSpans;
+  final Color backgroundColor;
+  final Color foregroundColor;
+
+  @override
+  String get id => messageId;
+
+  @override
+  Widget build(BuildContext context) {
+    final bannerMessagesController = Controllers.of(context).bannerMessages;
+    return Card(
+      color: backgroundColor,
+      child: Padding(
+        padding: const EdgeInsets.all(defaultSpacing),
+        child: Row(
+          children: <Widget>[
+            Expanded(
+              child: Container(
+                child: RichText(
+                  text: TextSpan(
+                    children: textSpans,
+                  ),
+                ),
+              ),
+            ),
+            const SizedBox(width: denseSpacing),
+            CircularIconButton(
```

Comment:
this icon looks quite large. is that intentional?

Reply:
Google's minimum recommended touch target size is around 48px.  WCAG suggests 44px or larger.


If the dismiss button had width/height of 24px, a smaller image imprint, the dismiss button should have additional padding to ensure the target size is within the 48px area.   

Going below the 24px size is not recommended for the dismiss button.  

This is a 24px with a 48px touch target for reference:
![24-close](https://user-images.githubusercontent.com/8105802/78840116-c8e5df80-79ae-11ea-9e40-1d50308335f8.png)